pyanaconda/display.py

```diff
@@ -276,77 +319,55 @@ def setup_display(anaconda, options):
             anaconda.display_mode = constants.DisplayModes.GUI
-        vnc_server.password = options.vncpassword
-
-        # Only consider vncconnect when vnc is a param
-        if options.vncconnect:
-            cargs = options.vncconnect.split(":")
-            vnc_server.vncconnecthost = cargs[0]
-            if len(cargs) > 1 and len(cargs[1]) > 0:
-                if len(cargs[1]) > 0:
-                    vnc_server.vncconnectport = cargs[1]
+        # FIXME: actually use user provided values
+        grd_server.rdp_username = options.rdp_username
+        grd_server.rdp_password = options.rdp_password
 
     if options.xdriver:
         write_xdriver(options.xdriver, root="/")
 
-    if anaconda.ksdata.vnc.enabled:
-        flags.usevnc = True
-        if not anaconda.gui_mode:
-            log.info("VNC requested via kickstart, switching Anaconda to GUI mode.")
-            anaconda.display_mode = constants.DisplayModes.GUI
-
-        if vnc_server.password == "":
-            vnc_server.password = anaconda.ksdata.vnc.password
-
-        if vnc_server.vncconnecthost == "":
-            vnc_server.vncconnecthost = anaconda.ksdata.vnc.host
-
-        if vnc_server.vncconnectport == "":
-            vnc_server.vncconnectport = anaconda.ksdata.vnc.port
-
     # check if GUI without WebUI
     if anaconda.gui_mode and not anaconda.is_webui_supported:
         mods = (tup[1] for tup in pkgutil.iter_modules(pyanaconda.ui.__path__, "pyanaconda.ui."))
         if "pyanaconda.ui.gui" not in mods:
             stdout_log.warning("Graphical user interface not available, falling back to text mode")
             anaconda.display_mode = constants.DisplayModes.TUI
-            flags.usevnc = False
-            flags.vncquestion = False
+            flags.use_rd = False
+            flags.rd_question = False
 
     # check if VNC can be started
     vnc_can_be_started, vnc_error_messages = check_vnc_can_be_started(anaconda)
     if not vnc_can_be_started:
         # VNC can't be started - disable the VNC question and log
         # all the errors that prevented VNC from being started
-        flags.vncquestion = False
+        flags.rd_question = False
         for error_message in vnc_error_messages:
             stdout_log.warning(error_message)
 
-    if anaconda.tui_mode and flags.vncquestion:
+    if anaconda.tui_mode and flags.rd_question:
         # we prefer vnc over text mode, so ask about that
         message = _("Text mode provides a limited set of installation "
                     "options. It does not offer custom partitioning for "
                     "full control over the disk layout. Would you like "
-                    "to use VNC mode instead?")
-        ask_vnc_question(anaconda, vnc_server, message)
+                    "to use remote graphical access via the RDP protocol instead?")
+        ask_vnc_question(anaconda, grd_server, message)
```

Comment:
You might want to rename this method to `ask_rd_question` or similar.

Reply:
Will do - there are actually a few remaining VNC references elsewhere I still like to remove/make generic. :)